src/stdlib_math_diff.fypp

```diff
@@ -12,27 +12,47 @@ contains
     #! `diff` computes differences of arrays of the ${t1}$ type.
     
     #:for k1, t1 in RI_KINDS_TYPES
-    pure module function diff_1_${k1}$(X, n) result(Y)
+    pure module function diff_1_${k1}$(X, n, prepend, append) result(Y)
         ${t1}$, intent(in) :: X(:)
         integer, intent(in), optional :: n
+        ${t1}$, intent(in), optional :: prepend(:), append(:)
         ${t1}$, allocatable :: Y(:)
+        integer :: size_prepend, size_append
         integer :: n_, i
 
+        size_prepend = 0
+        size_append = 0
         n_ = optval(n, 1)
+        if (present(prepend)) size_prepend = size(prepend) 
+        if (present(append)) size_append = size(append)
+        
+        if (size(X) <= n_) then
+            allocate(Y(size_prepend+size_append))
+            if (size_prepend > 0) Y(:size_prepend) = prepend(:)
+            if (size_append > 0) Y(size_prepend+1:) = append(:)
+            return
+        end if
 
         Y = X
         do i = 1, n_
             Y = Y(2:) - Y(:size(Y) - 1)
         end do
+        
+        if (size_prepend > 0) Y = [prepend(:), Y]
+        if (size_append > 0) Y = [Y, append(:)]
```

Comment:
I think that in this implementation, the meaning of `prepend` and `append` is different to in numpy (?). 

In numpy the idea is that we compute the standard `diff` of the array defined by `[prepend, X, append]`. So the output will have a size equal to `(-1 + size(prepend) + size(X) + size(append))`. 

Here is an illustration in numpy:
```python
import numpy
# Here is the standard diff, where the output_size is (input_size - 1)
numpy.diff(numpy.array([-2, -1, 0, 2, 3, 6, 10, 20]))
# array([ 1,  1,  2,  1,  3,  4, 10])

# Here we do the same thing, but the first 2 elements are set with prepend, and 
# the last 2 with append. It gives the same answer as above.
numpy.diff(numpy.array([0, 2, 3, 6]), prepend = [-2, -1], append=[10, 20])
# array([ 1,  1,  2,  1,  3,  4, 10])
```

Reply:
Sorry, I misunderstood it, now it's corrected.